src/circuit/gadgets/data.rs

```diff
@@ -201,17 +191,14 @@ impl<F: LurkField> GlobalAllocations<F> {
             Ok(Op2::Equal.as_field())
         })?;
 
-        let hash_sym = |name: &str| {
-            store
-                .get_lurk_sym(name, true)
-                .and_then(|s| store.hash_sym(s, HashScalar::Get))
-                .unwrap()
-        };
+        let c = store.get_constants();
 
         macro_rules! defsym {
-            ($var:ident, $name:expr) => {
-                let $var =
-                    AllocatedPtr::alloc_constant(&mut cs.namespace(|| $name), hash_sym($name))?;
+            ($var:ident, $name:expr, $cname:ident) => {
```

Comment:
IIUC, I think you are now repeating twice the same pattern-match, which makes the second rule unreachable. With De Bruijn indices:
```rust
            ($2: ident, , $1:expr, $0:ident) => {
                let $2 = AllocatedPtr::alloc_constant(
                    &mut cs.namespace(|| $1),
                    c.$0.scalar_ptr(),
                )?;
            };
            ($2:ident, $1:expr, $0:expr) => {
                let $2 = AllocatedPtr::alloc_constant(
                    &mut cs.namespace(|| $0),
                    hash_sym($1),
                )?;
            };
        }
```

Reply:
Right, I'll remove the second one. I'll need you to explain your replacement (but not now) since it's a more elaborate usage than I'm familiar with.

Reply:
oh, no, it's me abstracting away the names, nothing clever. Removing the second one is the right thing to do here.

Reply:
done